weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableQuery.ts

```diff
@@ -123,6 +84,72 @@ export const useCallsForQuery = (
   }, [callResults, calls.loading, total]);
 };
 
+export const useFilterSortby = (
+  filter: WFHighLevelCallFilter,
+  gridFilter: GridFilterModel,
+  gridSort: GridSortModel | undefined
+) => {
+  const sortBy = useDeepMemo(
+    useMemo(() => (gridSort ? getSortBy(gridSort) : []), [gridSort])
+  );
+  const lowLevelFilter: CallFilter = useMemo(() => {
+    return convertHighLevelFilterToLowLevelFilter(filter);
+  }, [filter]);
+  const filterByRaw = useMemo(() => getFilterByRaw(gridFilter), [gridFilter]);
+  const filterBy: Query | undefined = useMemo(
+    () => getFilterBy(filterByRaw),
+    [filterByRaw]
+  );
+
+  return {
+    sortBy,
+    lowLevelFilter,
+    filterBy,
+  };
+};
+
+const getFilterByRaw = (
+  gridFilter: GridFilterModel
+): Query['$expr'] | undefined => {
+  const setItems = gridFilter.items.filter(item => item.value !== undefined);
```

Comment:
This is kind of a confusing name, it's not a set object or a setter.

Reply:
I just factored it into a function, but I agree that is a confusing name...

Reply:
I'm going to rename as part of another fix - heads up for merge conflict: https://github.com/wandb/weave/pull/2082

Reply:
Yep thanks